main/test/src/mill/util/TestUtil.scala

```diff
@@ -43,13 +43,26 @@ object TestUtil extends MillTestKit {
    * controlled externally, so you can construct arbitrary dataflow graphs and
    * test how changes propagate.
    */
-  class TestTarget(inputs: Seq[Task[Int]], val pure: Boolean)(implicit ctx0: mill.define.Ctx)
-      extends Test(inputs)
-      with Target[Int] {
-    val ctx = ctx0.withSegments(ctx0.segments ++ Seq(ctx0.segment))
-    val readWrite = upickle.default.readwriter[Int]
+  class TestTarget(taskInputs: Seq[Task[Int]], val pure: Boolean)(implicit ctx0: mill.define.Ctx)
```

Comment:
This `TestTarget` was always a bit weird; unlike normal `TargetImpl`s, it didn't wrap a `Task[T]`, and instead implemented the `Task[T]` behavior itself, to keep the task graph simple for testing purposes. 

Now that we've simplified the `Task` hierarchy, we need to be a bit more explicit about implementing the task behavior ourselves, hence all these overrides